### PR TITLE
fix: Make MessageType#deletable return an accurate value.

### PR DESCRIPTION
**Pull request by RedDaedalus**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [x] Library interface (affecting end-user code) 
- [x] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

Fixes the `MessageType#deletable` getter.

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/api/entities/MessageType.java [hunk 1]
```diff
@@ -148,25 +148,31 @@
...
-    MessageType(int id, boolean system)
-    {
+    MessageType(int id, boolean system, boolean deletable) {
         this.id = id;
         this.system = system;
+        this.deletable = deletable;
+    }
+
+    MessageType(int id, boolean system) {
+        // most system messages are not deletable, most non-system messages are
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/entities/MessageType.java:169]: Pretty much every system message is deletable, the only one i know that prevents it is automod. Do you have actual exampels? I'm not a fan of having to say "true, true" on every enum constant here.

Comment (V-Play-Games) [reply src/main/java/net/dv8tion/jda/api/entities/MessageType.java:169]: Outside of Automod, Message sent in group DMs when changing the icon or the name, making calls, addition or removal of members are not deletable. But that's that

Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/api/entities/MessageType.java:169]: Group DMs are not even accessible to bots normally, so we shouldn't really worry about those types at all. We could comment them out instead. From what i can tell the only actually undeletable messages accessible to bots, are thread start messages and automod, am i wrong?

Comment (RedDaedalus) [reply src/main/java/net/dv8tion/jda/api/entities/MessageType.java:169]: Automod messages can be deleted. The two that can't be deleted are channel name and thread starter.

Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/api/entities/MessageType.java:169]: Then they changed it, they were not deletable when the type was added.

Comment (RedDaedalus) [reply src/main/java/net/dv8tion/jda/api/entities/MessageType.java:169]: Yes, that is what this PR is primarily addressing. 
image

File: src/main/java/net/dv8tion/jda/api/entities/MessageType.java [hunk 2]
```diff
@@ -148,25 +148,31 @@
...
+        this(id, system, !system);
+    }
+
+    MessageType(int id) {
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/entities/MessageType.java:173]: Remove these overload constructors and only have one with both booleans. I don't like having this implicit true and false here, its confusing.

File: src/main/java/net/dv8tion/jda/api/entities/MessageType.java [hunk 3]
```diff
@@ -148,25 +148,31 @@
...
+    private final boolean deletable;
 
-    MessageType(int id)
-    {
-        this(id, true); // true as default because the majority are system
-    }
-
-    MessageType(int id, boolean system)
-    {
+    MessageType(int id, boolean system, boolean deletable) {
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/entities/MessageType.java:162]:
```suggestion
    MessageType(int id, boolean system, boolean deletable)
    {
```

File: src/main/java/net/dv8tion/jda/api/entities/MessageType.java [hunk 4]
```diff
@@ -131,12 +131,12 @@
      * A new message sent as the first message in threads that are started from an existing message in the parent channel.
      * It only contains a message reference field that points to the message from which the thread was started.
      */
-    THREAD_STARTER_MESSAGE(21),
+    THREAD_STARTER_MESSAGE(21, false, true),
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/entities/MessageType.java:134]: I thought this one is not deletable?

File: src/main/java/net/dv8tion/jda/api/entities/MessageType.java [hunk 5]
```diff
@@ -193,13 +199,13 @@ public boolean isSystem()
 
     /**
      * Whether messages of this type can be deleted.
-     * <br>Currently the only type that cannot be deleted is {@link #AUTO_MODERATION_ACTION}.
+     * <br>This includes most system messages.
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/entities/MessageType.java:202]:
```suggestion
     * <br>These are messages which are required to stay such as thread starter messages.
```

**Review by MinnDevelopment — CHANGES_REQUESTED** [review]

**Review by MinnDevelopment — APPROVED** [review]

**Review by DV8FromTheWorld — APPROVED** [review]

**Pull request merged**